unit_tests/unit/test_keystore.py

```
@@ -0,0 +1,1020 @@
+
+    def test_removes_file_on_exception(self, mocked_s3):
+        captured = None
+        with pytest.raises(RuntimeError, match="boom"):
+            with materialize_ssh_key("scylla_test_id_ed25519") as key_path:
+                captured = key_path
+                raise RuntimeError("boom")
+        assert captured is not None and not captured.exists()
+
+    def test_prefers_tmpfs_when_available(self, mocked_s3):
+        """On Linux with /dev/shm available, keys land on tmpfs."""
+        if not os.path.isdir("/dev/shm"):
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/dev/shm"

(S108)